### PR TITLE
Lay the groundwork for nextgen events.

### PR DESCRIPTION
**Pull request by mikalstill**
I think we should be moving events to a scheme where they're queued through etcd, but not stored there. This would massively reduce our key count in etcd. Additionally, I see events and logs as increasingly overlapping. We should handle an event as a special case of a log message instead of a completely separate construct.

### CODE REVIEW

File: shakenfist/logutil.py [hunk 1]
```diff
@@ -141,8 +161,44 @@ def process(self, msg, kwargs):
...
+
+            encoded = json.dumps(
+                EventLogMessageTask(time.time(), msg, extra),
+                indent=4, sort_keys=True, cls=JSONEncoderTasks)
+            Etcd3Client().put(
+                '/sf/queues/eventlog/%s/%s-%s'
+                % (config.NODE_NAME, time.time(), util_random.random_id),
+                encoded, lease=None)
+        except Exception:
+            # Don't let event errors break logging
```
Comment (mandoonandy) [inline shakenfist/logutil.py:199]: Silent exceptions lead to not so silent pain. Any exceptions here need to surface somehow.

**Review by mandoonandy — COMMENTED** [review]
What's the overall intention here? Maybe a short design description is in order?

File: shakenfist/logutil.py (hunk 1 above)
Comment (mikalstill) [reply shakenfist/logutil.py:199]: My issue is that an exception here can cause logging to stop working entirely, which would stop you from seeing other exceptions...

**Comment by mikalstill** [comment]
Oh, this is against the wrong target branch of course.

**Pull request closed without merging**